reframe/utility/typecheck.py

```diff
@@ -342,6 +342,35 @@ def __rfm_cast_str__(cls, s):
         return s
 
 
+class Bool(metaclass=_BuiltinType):
```

Comment:
We need to document this properly for the users.

Reply:
Do you think I should document also for the use with external variables?

Reply:
Yes, we should mention that if you want to define a boolean variable that can be properly set from the command line, you should define it as `Bool`.